python/seldon_core/batch_processor.py

```diff
@@ -224,10 +256,133 @@ def _start_request_worker(
+    requests to be combined into a single prediction.
+    Parameters
+    ---
+    input_data
+        The input data containing the indexes, instance_ids and predictions
+    data_type
+        The data type to send which can be str, json and data
+    sc
+        The instance of SeldonClient to use to send the requests to the seldon model
+    retries
+        The number of times to retry the request
+    batch_id
+        The unique identifier for the batch which is passed to all requests
+    Returns
+    ---
+        A string serialised result of the response (or equivalent data with error info)
+    """
+
+    indexes = [x[0] for x in input_data]
+    instance_ids = [x[1] for x in input_data]
+    first_prediction = input_data[0][2]
+
+    predict_kwargs = {}
+    meta = {
+        "tags": {
+            "batch_id": batch_id,
+        }
+    }
+
+    predict_kwargs["meta"] = meta
+    predict_kwargs["headers"] = {"Seldon-Puid": instance_ids[0]}
+
+    try:
+        if data_type == "data":
+            # Initialise concatenated array for data
+            data = json.loads(first_prediction)
+            data_np = np.array(data)
+            concat = data_np
+            for i, raw_data in enumerate(input_data):
+                # Already added first item.
+                if i == 0:
+                    continue
+                data = json.loads(raw_data[2])
+                data_np = np.array(data)
+                concat = np.concatenate((concat, data_np))
+            predict_kwargs["data"] = concat
+        data = json.loads(first_prediction)
+        if data_type == "str":
+            predict_kwargs["str_data"] = data
+        elif data_type == "json":
+            predict_kwargs["json_data"] = data
+
+        response = None
+        for i in range(retries):
+            try:
+                seldon_payload = sc.predict(**predict_kwargs)
+                assert seldon_payload.success
+                response = seldon_payload.response
+                break
+            except (requests.exceptions.RequestException, AssertionError) as e:
+                logger.error(f"Exception: {e}, retries {retries}")
+                if i == (retries - 1):
+                    raise
+
+    except Exception as e:
+        error_resp = {
+            "status": {"info": "FAILURE", "reason": str(e), "status": 1},
+            "meta": meta,
+        }
+        print("Exception: %s" % e)
+        str_output = json.dumps(error_resp)
+        return [str_output]
+
+    # Take the response create new responses for each request
+    responses = []
+    # If tensor then prepare the ndarray
+    tensor_ndarray = np.array(())
+    if payload_type == "tensor":
+        tensor = np.array(response["data"]["tensor"]["values"])
+        shape = response["data"]["tensor"]["shape"]
+        tensor_ndarray = tensor.reshape(shape)
+
+    for i in range(len(input_data)):
+        new_response = copy.deepcopy(response)
```

Comment:
How inefficient is this copy for large batch size?

Reply:
I believe it should have negligible impact when considering we still need to read and write each row separately to the file, but I agree it's not ideal.

We need to split the single response into X responses which share 90% of the same content. I can't think of a more efficient way to do it off the top of my head?

Reply:
Why do you need to deepcopy anyway? Can't you just save 'response["data"]["ndarray"]' as a variable and then always just override the original value? Eventually you dump them into strings so overriding won't really "ruin" the ones you are already done with, right?

Reply:
Yeah this is a good idea, would you like to raise a PR for it @naor2013? If not I can address at a later date.